tools/dependency/tests/test_pip_check.py

```diff
@@ -31,51 +31,68 @@ def test_pip_checker_config_requirements():
             == [('updates',), {}])
 
 
-def test_pip_checker_dependabot_config(patches):
+@pytest.mark.parametrize("isdict", [True, False])
+def test_pip_checker_dependabot_config(patches, isdict):
     checker = pip_check.PipChecker("path1", "path2", "path3")
     patched = patches(
         "utils",
         ("PipChecker.path", dict(new_callable=PropertyMock)),
-        "os.path.join",
         prefix="tools.dependency.pip_check")
 
-    with patched as (m_utils, m_path, m_join):
-        assert checker.dependabot_config == m_utils.from_yaml.return_value
+    with patched as (m_utils, m_path):
+        if isdict:
+            m_utils.from_yaml.return_value = {}
+
+        if isdict:
+            assert checker.dependabot_config == m_utils.from_yaml.return_value
+        else:
+
```

Comment:
I'm thinking we're going to have to agree to disagree on this style, but each time I see it I find i very jarring..

Reply:
k - i give in - ill remove the preceding ws from the contexts

it jars with me also, but at the same time ive read so much code that has a context buried deep in it that is not obvious what is going on

Reply:
(i guess also a dislike of ~old-js style nested closures)